### PR TITLE
feat(ST-89): fetch classroom details using jwt token of a teacher

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Added endpoint for fetching classroom details for specific teacher.
- Added function in the classroom service for fetching the classroom details.
## Associated Ticket Link(s)

- https://trello.com/c/g4DE1JJS

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -56,4 +56,36 @@ export class ClassroomsService {
...
+
+  async getClassroomsForUser(userId: number): Promise<ClassroomResponseDto[]> {
+    const user = await this.em.findOne(User, { id: userId }, { populate: ["teacher"] });
+    if (!user) {
+      throw new NotFoundException("User not found");
+    }
+
+    if (!user.teacher) {
+      throw new ForbiddenException("Only teachers can view classrooms");
+    }
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts]: Both teachers and students should be able to view classrooms.

File: src/classrooms/classrooms.controller.ts [hunk 2]
```diff
@@ -19,4 +19,9 @@ export class ClassroomsController {
     const userId = req.user.id;
     return this.classroomsService.createClassroom(createClassroomDto, userId);
   }
+
+  @Get()
+  getClassrooms(@Req() req: AuthenticatedRequest): Promise<ClassroomResponseDto[]> {
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.controller.ts]: Best to use @CurrentUser decorator here

File: src/classrooms/classrooms.service.ts [hunk 3]
```diff
@@ -56,4 +56,36 @@ export class ClassroomsService {
...
+      (classroom) =>
+        new ClassroomResponseDto({
+          id: classroom.id,
+          title: classroom.title,
+          subject: classroom.subject,
+          classTime: classroom.classTime,
+          days: classroom.days,
+          userId: user.id,
+        }),
+    );
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts]: Is this doing what a serializer should do?

File: src/classrooms/classrooms.service.ts [hunk 4]
```diff
@@ -56,4 +56,36 @@ export class ClassroomsService {
...
+    }
+
+    const classrooms = await this.em.find(
+      Classroom,
+      { teacher: user.teacher },
+      {
+        orderBy: { classTime: "DESC" },
+        populate: ["teacher"],
+      },
+    );
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:64]: Probably need to modify find queries based on role

Comment (ghost) [reply src/classrooms/classrooms.service.ts:64]: Bhaiya, I will do that once I start working with the student enrollment table. For now I am only working with teacher creating a classroom. For the moment I am leaving a TODO comment

File: src/classrooms/classrooms.service.ts (hunk 3 above)
Comment (ghost) [reply src/classrooms/classrooms.service.ts]: Yes bhaiya, it's taking the raw classroom entity and transforming it into a response format suitable for frontend/client.

Comment (ishmam-iqbal-sazim) [reply src/classrooms/classrooms.service.ts]: Mikro ORM provides a serialize method to do this. You can use that.

And serializer should be called at the controller level imported from classrooms.serializer.ts.

File: src/classrooms/classrooms.serializer.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    classTime: wrappedClassroom.classTime,
+    days: wrappedClassroom.days,
+    userId: userId,
+  };
+};
+
+export const serializeClassrooms = (
+  classrooms: Classroom[],
+  userId: number,
+): ClassroomResponseDto[] => classrooms.map((classroom) => serializeClassroom(classroom, userId));
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.serializer.ts]: Should be a class which is injected into the module. Please look into the template for the structure.

Should be at least 2 methods present here serializeOne and serializeMany

File: src/classrooms/classrooms.controller.ts [hunk 6]
```diff
@@ -22,4 +23,10 @@ export class ClassroomsController {
   ): Promise<ClassroomResponseDto> {
     return this.classroomsService.createClassroom(createClassroomDto, user.id);
   }
+
+  @Get()
+  async getClassrooms(@CurrentUser() user: { id: number }): Promise<ClassroomResponseDto[]> {
+    const classrooms = await this.classroomsService.getClassroomsForUser(user.id);
+    return serializeClassrooms(classrooms, user.id);
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.controller.ts]:
```suggestion
    return this.classroomsSerializer.serializeMany(classrooms);
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**